src/routes/transactions/transactions.service.ts

```diff
@@ -175,4 +182,114 @@ export class TransactionsService {
       results,
     };
   }
+
+  async getTransactionQueue(
+    chainId: string,
+    routeUrl: Readonly<URL>,
+    safeAddress: string,
+    paginationData: PaginationData,
+  ): Promise<Page<QueuedItem>> {
+    const pagination = this.getAdjustedPagination(paginationData);
+    const safeInfo = await this.safeRepository.getSafe(chainId, safeAddress);
+    const transactions = await this.safeRepository.getTransactionQueueByNonce(
+      chainId,
+      safeAddress,
+      pagination.limit,
+      pagination.offset,
+    );
+
+    const { next, previous } = transactions;
+    const nextURL = cursorUrlFromLimitAndOffset(routeUrl, next);
```

Comment:
I'm working in something similar. I think that here you should consider the user request offset and limit instead the adjusted pagination, the tx-service response is calculated from the modified values.

Reply:
Yes, you're right, the offset and limit are not being properly calculated due to this. We need to calculate them with the original values instead of relying on the transaction service response, I'll fix it, thanks!